services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -4700,7 +4700,21 @@ public boolean containsHelixResource(String clusterName, String kafkaTopic) {
   @Override
   public void deleteHelixResource(String clusterName, String kafkaTopic) {
     checkControllerLeadershipFor(clusterName);
-    helixAdminClient.dropResource(clusterName, kafkaTopic);
+    List<String> instances = getStorageNodes(clusterName);
+
+    for (String instance: instances) {
+      Map<String, List<String>> disabledPartitions =
+          getHelixAdminClient().getDisabledPartitionsMap(clusterName, instance);
+      for (Map.Entry<String, List<String>> entry: disabledPartitions.entrySet()) {
+        if (entry.getKey().equals(kafkaTopic)) {
+          // clean up disabled partition map, so that it does not grow indefinitely with dropped resources
+          getHelixAdminClient().enablePartition(true, clusterName, instance, kafkaTopic, entry.getValue());
+          LOGGER.info("Cleaning up disabled replica of resource {}, partitions {}", entry.getKey(), entry.getValue());
+        }
+      }
+    }
+
+    getHelixAdminClient().dropResource(clusterName, kafkaTopic);
```

Comment:
Does the order matter?
Such as dropping resource first before re-enabling these disabled partition?
If we switch the order, will re-enabling disabled partition operation still work?
I am worried that the re-enabling before dropping could trigger unnecessary rebalancing.

Reply:
I find the change and the description confusing. Can you please clarify how enabling the partition helps to clean up (remove?) the map? Also wouldn't it make sense to make it part of HelixAdminClient rather than VeniceHelixAdmin?

Reply:
The disabled partitions are kept in a config map for a resource on a znode.  If we enable those map, it clears those map. The API call to helix is part of `HelixAdminClient`, but the rest of the logic should stay in VeniceHelixAdmin